go/pkg/client/bytestream_test.go

```diff
@@ -0,0 +1,295 @@
+		},
+		{
+			description: "not found",
+			ls:          nil,
+			data:        logStreamData,
+			opts:        []ByteStreamWriteOption{ByteStreamOptOffset(0), ByteStreamOptFinishWrite(false)},
+		},
+		{
+			description: "invlid smaller offset",
+			ls:          &logStream{logicalOffset: 4},
+			data:        logStreamData,
+			opts:        []ByteStreamWriteOption{ByteStreamOptOffset(1), ByteStreamOptFinishWrite(false)},
+		},
+		{
+			description: "invlid larger offset",
+			ls:          &logStream{logicalOffset: 2},
+			data:        logStreamData,
+			opts:        []ByteStreamWriteOption{ByteStreamOptOffset(4), ByteStreamOptFinishWrite(false)},
+		},
+		{
+			description: "invlid negative offset",
+			ls:          &logStream{logicalOffset: 0},
+			data:        logStreamData,
+			opts:        []ByteStreamWriteOption{ByteStreamOptOffset(-1), ByteStreamOptFinishWrite(false)},
+		},
+	}
+
+	b := newServer(t)
+	defer b.shutDown()
+	for _, test := range tests {
+		t.Run(test.description, func(t *testing.T) {
+			lsID := test.description
+			if test.ls != nil {
+				test.ls.logStreamID = lsID
+				b.fake.logStreams[lsID] = test.ls
+			}
+			data := []byte("Hello World!")
+			ChunkMaxSize(len(data)).Apply(b.client)
+			writtenBytes, err := b.client.WriteBytesWithOptions(b.ctx, lsID, data, test.opts...)
+			if err == nil {
+				t.Error("WriteBytesWithOffset() got nil error, want non-nil error")
+			}
+			if writtenBytes != 0 {
+				t.Errorf("WriteBytesWithOffset() got %d byte(s), want 0 byte", writtenBytes)
+			}
+		})
+	}
+}
+
+func TestWirteBytesWithOptions_LogStream_FinishWrite(t *testing.T) {
+	b := newServer(t)
+	defer b.shutDown()
+	dataSize := len(logStreamData)
+	for i, flag := range []ByteStreamWriteOption{ByteStreamOptFinishWrite(true), ByteStreamOptFinishWrite(false)} {
+		t.Run(fmt.Sprintf("ByteStreamOptFinishWrite=%t", flag), func(t *testing.T) {
+			lsID := fmt.Sprintf("logstream%d", i+1)
+			b.fake.logStreams[lsID] = &logStream{logStreamID: lsID, logicalOffset: 0}
+			ChunkMaxSize(dataSize).Apply(b.client)
+
+			_, err := b.client.WriteBytesWithOptions(b.ctx, lsID, logStreamData, flag)
+			if err != nil {
+				t.Errorf("WriteBytesWithOffset() failed unexpectedly: %v", err)
+			}
+
+			if b.fake.logStreams[lsID].logicalOffset != int64(dataSize) {
+				t.Errorf("WriteBytesWithOffset() = %d, want %d", b.fake.logStreams[lsID].logicalOffset, int64(dataSize))
+			}
+			if b.fake.logStreams[lsID].finalized != bool(flag.(ByteStreamOptFinishWrite)) {
+				t.Errorf("WriteBytesWithOffset() got %t, want %t state", b.fake.logStreams[lsID].finalized, bool(flag.(ByteStreamOptFinishWrite)))
+			}
+		})
+
+	}
+}
+
+func TestWirteBytesWithOptions_LogStream_Offset(t *testing.T) {
+	b := newServer(t)
+	defer b.shutDown()
+	dataSize := len(logStreamData)
+	for _, offset := range []ByteStreamWriteOption{ByteStreamOptOffset(0), ByteStreamOptOffset(6)} {
+		t.Run(fmt.Sprintf("ByteSteramOptFinishWrite=%t", offset), func(t *testing.T) {
```

Comment:
Typo
```suggestion
		t.Run(fmt.Sprintf("ByteStreamOptFinishWrite=%t", offset), func(t *testing.T) {
```

Reply:
Removed the unit test method as the implementation has changed a bit.